src/huggingface_hub/file_download.py

```diff
@@ -961,6 +961,31 @@ def repo_folder_name(*, repo_id: str, repo_type: str) -> str:
     return REPO_ID_SEPARATOR.join(parts)
 
 
+def _check_disk_space(expected_size: int, target_dir: Union[str, Path]) -> bool:
+    """Check disk usage and log a warning if there is not enough disk space to download the file.
+
+    Args:
+        expected_size (`int`):
+            The expected size of the file in bytes.
+        target_dir (`str`):
+            The directory where the file will be stored after downloading.
+    """
+
+    target_dir = str(target_dir)
+    target_dir_free = shutil.disk_usage(target_dir).free
+
+    has_enough_space = target_dir_free >= expected_size
+
+    if not has_enough_space:
+        logger.warning(
```

Comment:
Ah yes! You should use `warnings.warn` here for a proper warning message. This is different than logger.warning logs. Glad that you've implemented tests to realize that!

Reply:
Ok, if `warnings.warn` is okay with you, it becomes much easier.

However, I'm at a loss on how to test for no warning with `self.assertWarns`.

I propose the following (with `import warnings` earlier in the file):
```python
class TestDiskUsageWarning(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        # Test with 100MB expected file size
        cls.expected_size = 100 * 1024 * 1024

    @patch("huggingface_hub.file_download.shutil.disk_usage")
    def test_disk_usage_warning(self, disk_usage_mock: Mock) -> None:
        # Test with only 1MB free disk space / not enough disk space,  with UserWarning expected
        disk_usage_mock.return_value.free = 1024 * 1024
        with warnings.catch_warnings(record=True) as w:
            # Cause all warnings to always be triggered.
            warnings.simplefilter("always")
            _check_disk_space(expected_size=self.expected_size, target_dir=disk_usage_mock)
            assert len(w) == 1
            assert issubclass(w[-1].category, UserWarning)

        # Test with 200MB free disk space / enough disk space,  with no warning expected
        disk_usage_mock.return_value.free = 200 * 1024 * 1024
        with warnings.catch_warnings(record=True) as w:
            # Cause all warnings to always be triggered.
            warnings.simplefilter("always")
            _check_disk_space(expected_size=self.expected_size, target_dir=disk_usage_mock)
            assert len(w) == 0
```

Reply:
Really good way to test the absence of warnings! So yes, go ahead with this solution since it works :)